From 281fec37f578108077346b3e1b2b0d3f1b0efa2b Mon Sep 17 00:00:00 2001
From: Robert Bragg <robert@linux.intel.com>
Date: Mon, 22 Aug 2011 02:16:36 +0100
Subject: [PATCH 5/7] utils: avoid redundant set/endpwent around getpwuid

set/endpwent are only required for iterating through passwd entries
using getpwent(). Since we are explicitly requesting a passwd entry for
a uid then the set/endpwent calls are redundant.

Removing these redundant calls is required for building on Android since
their C library doesn't implement these.
---
 glib/gutils.c |    6 +-----
 1 files changed, 1 insertions(+), 5 deletions(-)

Index: glib-2.30.2/glib/gutils.c
===================================================================
--- glib-2.30.2.orig/glib/gutils.c
+++ glib-2.30.2/glib/gutils.c
@@ -1853,11 +1853,7 @@ g_get_any_init_do (void)
 #  endif /* HAVE_POSIX_GETPWUID_R || HAVE_NONPOSIX_GETPWUID_R */
     
     if (!pw)
-      {
-	setpwent ();
-	pw = getpwuid (getuid ());
-	endpwent ();
-      }
+      pw = getpwuid (getuid ());
     if (pw)
       {
 	g_user_name = g_strdup (pw->pw_name);
